presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2833,8 +2835,13 @@ private Session createViewSession(Optional<String> catalog, Optional<String> sch
                 .setRemoteUserAddress(session.getRemoteUserAddress().orElse(null))
                 .setUserAgent(session.getUserAgent().orElse(null))
                 .setClientInfo(session.getClientInfo().orElse(null))
-                .setStart(session.getStart())
-                .build();
+                .setStart(session.getStart());
+
+        for (Entry<String, String> entry : session.getSystemProperties().entrySet()) {
```

Comment:
what about catalog session properties?

Reply:
they probably should be copied too, but the relationship between catalog properties and unprocessed catalog properties is not clear to me and I did not have the time to poke in the insides of the session to understand how they relate to each other. 

Ideally, the session would support a `toSessionBuilder()` which prepopulates a new session from an existing one but as this is literally the only place where it is used, it might be overkill.

Reply:
How about using `io.prestosql.Session#builder(io.prestosql.Session)` and override properties that needs to be overwritten?

> catalog properties and unprocessed catalog properties

I think we need them both.